src/Gaffer/Reference.cpp

```diff
@@ -173,19 +173,68 @@ class Reference::PlugEdits : public boost::signals::trackable
 			return edit->metadataEdits.find( key ) != edit->metadataEdits.end();
 		}
 
+		bool isChildEdit( const Plug *plug ) const
+		{
+			const Plug *parent = plug->parent<Plug>();
+			if( !parent )
+			{
+				return false;
+			}
+
+			const PlugEdit *edit = plugEdit( parent );
+			if( !edit )
+			{
+				return false;
+			}
+
+			if( edit->sizeAfterLoad == -1 || parent->children().size() <= (size_t)edit->sizeAfterLoad )
+			{
+				return false;
+			}
+
+			// Conceptually we want to compare the index of `plug` against
+			// `sizeAfterLoad`. But finding the index currently requires linear
+			// search. We expect the UI to only allow creation of new plugs in
+			// originally-empty containers (to avoid merge hell on reload),
+			// meaning that `sizeAfterLoad` can be expected to be either 0 or 1
+			// (the latter for RowsPlug with a default row). So it is quicker to
+			// reverse the test and search for plug in the range `[0,
+			// sizeAfterLoad)`.
+			return !std::any_of(
+				parent->children().begin(), parent->children().begin() + edit->sizeAfterLoad,
+				[plug]( const GraphComponentPtr &child ) { return child == plug; }
+			);
+		}
+
 		void transferEdits( Plug *oldPlug, Plug *newPlug ) const
 		{
 			transferEditedMetadata( oldPlug, newPlug );
+			// Transfer additional children.
+			if( auto *edit = plugEdit( oldPlug ) )
+			{
+				if( edit->sizeAfterLoad != -1 )
+				{
+					while( oldPlug->children().size() > (size_t)edit->sizeAfterLoad )
```

Comment:
This while statement has me completely baffled.  I was understanding sizeAfterLoad as the size after loading the stored reference, before any edits are applied - so the whole point would be that it wouldn't change size while transferring edits?  But this relies on sizeAfterLoad increasing as edits are transferred?  At least, I can't see any way that oldPlug->children() would be shrinking?  I feel like I'm badly missing something.

Reply:
Oh, wait, of course - addChild is actually reparenting and stealing the children of oldChild.  It is called transferEdits, not copyEdits, I guess this all makes sense.

Reply:
Yep, you've got it. Perhaps there is a way of formulating this that feels a little more natural or obvious, but I couldn't find one, at least not without taking a copy of `oldPlug->children()` to work with. Added a clarifying comment in bba39a1.